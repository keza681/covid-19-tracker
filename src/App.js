import { Routes, Route } from 'react-router-dom';
import React from 'react';
import NavBar from './components/Navbar';
import HomePage from './components/HomePage';
import Details from './components/Details';

function App() {
  return (
    <div>
      <NavBar />
      <Routes>
        <Route path="/" element={<HomePage />} />
        <Route path="/:id" element={<Details />} />
      </Routes>
    </div>

  );
}

export default App;
