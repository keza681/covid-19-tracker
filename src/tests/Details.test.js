/* eslint-disable camelcase */
import React from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import configureStore from '../redux/store';
import App from '../App';

describe('Test of detail page component', () => {
  test('It should render the details successfully', () => {
    const tree = renderer
      .create(
        <Provider store={configureStore}>
          <BrowserRouter>
            <App />
          </BrowserRouter>
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
