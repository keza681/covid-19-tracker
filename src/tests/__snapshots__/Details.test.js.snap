// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Countries snapshot renders correctly 1`] = `
<div>
  <nav>
    <section
      className="navLinks"
    >
      <div>
        <a
          aria-current="page"
          className="active"
          href="/"
          onClick={[Function]}
        >
          <img
            alt="back"
            src="https://img.icons8.com/ios-filled/24/ffffff/back.png"
          />
        </a>
      </div>
      <div
        className="navIcons"
      >
        <img
          alt="mic"
          className="navIcon"
          src="https://img.icons8.com/fluency-systems-filled/24/ffffff/microphone.png"
        />
        <img
          alt="settings"
          className="navIcon"
          src="https://img.icons8.com/external-tanah-basah-glyph-tanah-basah/24/ffffff/external-setting-essentials-pack-tanah-basah-glyph-tanah-basah.png"
        />
      </div>
    </section>
  </nav>
  <div>
    <header
      className="cover"
    >
      <div
        className="title"
      >
        <div
          className="logo"
        >
          <div>
            <h1>
              Covid Cases
            </h1>
          </div>
        </div>
      </div>
    </header>
    <div
      className="search"
    >
      <div
        className="myDate"
      >
        <h2>
           Stats by country
        </h2>
        <h3
          className="date"
        >
          2022-04-28
        </h3>
      </div>
      <input
        className="inputSearch"
        id="search"
        name="search"
        onChange={[Function]}
        placeholder="search here"
        type="text"
        value=""
      />
    </div>
    <ul
      className="myCounty"
    />
  </div>
</div>
`;

exports[`Test of detail page component It should render the details successfully 1`] = `
<div>
  <nav>
    <section
      className="navLinks"
    >
      <div>
        <a
          aria-current="page"
          className="active"
          href="/"
          onClick={[Function]}
        >
          <img
            alt="back"
            src="https://img.icons8.com/ios-filled/24/ffffff/back.png"
          />
        </a>
      </div>
      <div
        className="navIcons"
      >
        <img
          alt="mic"
          className="navIcon"
          src="https://img.icons8.com/fluency-systems-filled/24/ffffff/microphone.png"
        />
        <img
          alt="settings"
          className="navIcon"
          src="https://img.icons8.com/external-tanah-basah-glyph-tanah-basah/24/ffffff/external-setting-essentials-pack-tanah-basah-glyph-tanah-basah.png"
        />
      </div>
    </section>
  </nav>
  <div>
    <header
      className="cover"
    >
      <div
        className="title"
      >
        <div
          className="logo"
        >
          <div>
            <h1>
              Covid Cases
            </h1>
          </div>
        </div>
      </div>
    </header>
    <div
      className="search"
    >
      <div
        className="myDate"
      >
        <h2>
           Stats by country
        </h2>
        <h3
          className="date"
        >
          2022-04-28
        </h3>
      </div>
      <input
        className="inputSearch"
        id="search"
        name="search"
        onChange={[Function]}
        placeholder="search here"
        type="text"
        value=""
      />
    </div>
    <ul
      className="myCounty"
    />
  </div>
</div>
`;
